### PR TITLE
Add ChartDefaultURL to support custom rancher-charts repo in testing

### PR DESCRIPTION
**Pull request by manno**
## Problem

When testing different fleet builds in Rancher, we need to set up a fork of rancher/charts (described here: https://github.com/rancher/fleet/blob/master/DEVELOPING.md#fleet-in-rancher---using-a-custom-fork).

However we don't want to (should not) create a test branch in rancher/charts directly, so we have to change the rancher code and change the [hard coded URL](https://github.com/manno/rancher/blob/release/v2.7/pkg/data/dashboard/repo.go#L29).
 
## Solution

This adds a setting for `ChartDefaultURL` (env var `CATTLE_CHART_DEFAULT_URL`), which will be used for rancher-charts when the controller is started and adds its repos.

## Testing

Since we only use this variable for dev purposes, everything should work as before.

### CODE REVIEW

**Review by raulcabello — APPROVED** [review]

**Review by cbron — COMMENTED** [review]
Unless this is isolated fleet code, please get reviews from core rancher team before merging. I would like to continue this PR discussion to see if this is what we actually want to do.

File: pkg/data/dashboard/repo.go
Comment (manno) [reply pkg/data/dashboard/repo.go:23]: How do we find out who owns code? Is there something like a Codeowners file?

File: pkg/settings/setting.go
Comment (manno) [reply pkg/settings/setting.go:113]: We also want to test released Rancher versions with custom fleet versions. Ideally in the future that would be possible without a repo, or with a local repo. Maybe via the bundled system catalog. That would make the setup for automated tests easier.

For now, if it's ok to use a fleet test branch in rancher/charts that would be good.

**Comment by manno** [comment]
I think it's best to revert this merge.

**Pull request merged**
